yjit/src/core.rs

```diff
@@ -2143,9 +2172,11 @@ fn remove_from_graph(blockref: &BlockRef) {
         let mut pred_branch = pred_branchref.borrow_mut();
 
         // If this is us, nullify the target block
-        for pred_succ in pred_branch.targets.iter_mut().flatten() {
-            if pred_succ.block.as_ref() == Some(blockref) {
-                pred_succ.block = None;
+        for target_idx in 0..=1 {
+            if let Some(target) = pred_branch.targets[target_idx].as_ref() {
+                if target.get_block().as_ref() == Some(blockref) {
+                    pred_branch.targets[target_idx] = None;
```

Comment:
So, I remember considering doing this when I first introduced `BranchTarget` and I got scared that this state where `.block=None, but .address=Some...` is important. I'd say it's a very very small risk and it's definitely a good time to take such risk right now. In case it does end up being important, could you make a note of this in the commit log? We should just keep this in the back of our mind in case it come up; we'll have to rely on memory for this one.

Reply:
Sure, I'll make a note of this in the commit message when squash-merging this.

Reply:
summarized my intention in the commit message of https://github.com/ruby/ruby/commit/5ce0c13f181d7b2d50637758894bcbc357b0ce65. Let me know if I'm missing something.